contracts/0.8.9/oracle/AccountingOracle.sol

```diff
@@ -0,0 +1,817 @@
+    // only called before the migration
+
+    function getBeaconSpec() external view returns (
+        uint64 epochsPerFrame,
+        uint64 slotsPerEpoch,
+        uint64 secondsPerSlot,
+        uint64 genesisTime
+    );
+
+    function getLastCompletedEpochId() external view returns (uint256);
+
+    // only called after the migration
+
+    function handleConsensusLayerReport(
+        uint256 refSlot,
+        uint256 clBalance,
+        uint256 clValidators
+    ) external;
+}
+
+
+interface IStakingRouter {
+    function getExitedKeysCountAcrossAllModules() external view returns (uint256);
+
+    function updateExitedKeysCountByStakingModule(
+        uint256[] calldata _moduleIds,
+        uint256[] calldata _exitedKeysCounts
+    ) external;
+
+    function reportStakingModuleExitedKeysCountByNodeOperator(
+        uint256 _stakingModuleId,
+        uint256[] calldata _nodeOperatorIds,
+        uint256[] calldata _exitedKeysCounts
+    ) external;
+
+    function reportStakingModuleStuckKeysCountByNodeOperator(
+        uint256 _stakingModuleId,
+        uint256[] calldata _nodeOperatorIds,
+        uint256[] calldata _stuckKeysCounts
+    ) external;
+}
+
+
+interface IWithdrawalQueue {
+    function updateBunkerMode(bool isBunkerMode, uint256 prevReportTimestamp) external;
+}
+
+
+contract AccountingOracle is BaseOracle {
+    using ResizableArray for ResizableArray.Array;
+    using UnstructuredStorage for bytes32;
+    using SafeCast for uint256;
+
+    error ComponentLocatorCannotBeZero();
+    error AdminCannotBeZero();
+    error LegacyOracleCannotBeZero();
+    error IncorrectOracleMigration(uint256 code);
+    error SenderNotAllowed();
+    error InvalidExitedValidatorsData();
+    error NumExitedValidatorsCannotDecrease();
+    error ExitedValidatorsLimitExceeded(uint256 limitPerDay, uint256 exitedPerDay);
+    error UnsupportedExtraDataFormat(uint256 format);
+    error UnsupportedExtraDataType(uint256 dataType);
+    error MaxExtraDataItemsCountExceeded(uint256 maxItemsCount, uint256 receivedItemsCount);
+    error CannotSubmitExtraDataBeforeMainData();
+    error ExtraDataAlreadyProcessed();
+    error ExtraDataListOnlySupportsSingleTx();
+    error UnexpectedExtraDataFormat(uint256 expectedFormat, uint256 receivedFormat);
+    error UnexpectedExtraDataItemsCount(uint256 expectedCount, uint256 receivedCount);
+    error UnexpectedExtraDataIndex(uint256 expectedIndex, uint256 receivedIndex);
+    error InvalidExtraDataSortOrder();
+
+    event DataBoundariesSet(uint256 maxExitedValidatorsPerDay, uint256 maxExtraDataListItemsCount);
+    event ExtraDataSubmitted(uint256 indexed refSlot, uint256 itemsProcessed, uint256 itemsCount);
+
+    event WarnExtraDataIncompleteProcessing(
+        uint256 indexed refSlot,
+        uint256 processedItemsCount,
+        uint256 itemsCount
+    );
+
+    struct DataBoundaries {
+        uint64 maxExitedValidatorsPerDay;
+        uint64 maxExtraDataListItemsCount;
```

Comment:
is it how much chunks (currently only one chunk) or number of items in the list?

Reply:
can we move this to secutriy checks module or keep here?

Reply:
done